autoencoder.py

```diff
@@ -0,0 +1,46 @@
+from torch import nn
+
+"""
+Autoencoder
+
+This model is currently used to reduce the dimensionality of the image vectors produced by the ResNet50 model.
+Thhis model has, in practice, proven itself to be more effective in our image inference tasks
```

Comment:
Maybe update this comment to reflect our most recent understanding of ResNet50 vs vit